cmd/jaeger/config-elasticsearch.yaml

```diff
@@ -17,6 +17,13 @@ extensions:
       some_storage:
         elasticsearch:
           index_prefix: "jaeger-main"
+          index_date_layout_spans: "2006-01-02"
```

Comment:
I want to use the opportunity of starting a brand new configuration scheme to structure it better. There are different configuration options for each index type that can be much better represented as grouped under one name, e.g.
```yaml
elasticsearch:
  indices:
    services:
      date_format: "2006-01-02"
      rollover_frequency: "hour"
    sampling:
      ...
```

Reply:
I also prefer this structured configuration, I tried a version, please review~